### PR TITLE
Add plugin warnings

### PR DESCRIPTION
**Pull request by julianoes**
By oversight we did not have the special warnings enabled for the plugins.

This enables the warnings again and fixes a few that crept in. And one was actually a bug so that confirms that warnings are important!

### CODE REVIEW

File: plugins/calibration/calibration_impl.cpp [hunk 1]
```diff
@@ -330,7 +330,7 @@ void CalibrationImpl::process_statustext(const mavlink_message_t &message)
             // FALLTHROUGH
         case CalibrationStatustextParser::Status::CANCELLED:
             _calibration_callback = nullptr;
-            _state == State::NONE;
+            _state = State::NONE;
```
Comment (JonasVautherin) [inline plugins/calibration/calibration_impl.cpp]: Good catch :)

File: plugins/telemetry/telemetry.cpp [hunk 2]
```diff
@@ -319,17 +319,17 @@ const char *Telemetry::result_str(Result result)
...
 {
-    return abs(lhs.position.north_m - rhs.position.north_m) <=
-               std::numeric_limits<double>::epsilon() &&
-           abs(lhs.position.east_m - rhs.position.east_m) <=
-               std::numeric_limits<double>::epsilon() &&
-           abs(lhs.position.down_m - rhs.position.down_m) <=
+    return fabs(lhs.position.north_m - rhs.position.north_m) <=
                std::numeric_limits<float>::epsilon() &&
-           abs(lhs.velocity.north_m_s - rhs.velocity.north_m_s) <=
+           fabs(lhs.position.east_m - rhs.position.east_m) <=
```
Comment (JonasVautherin) [inline plugins/telemetry/telemetry.cpp]: Right.

File: core/global_include.h [hunk 3]
```diff
@@ -5,14 +5,15 @@
 #include <chrono>
 #include <thread>
 
-#ifdef WINDOWS
-#ifndef _USE_MATH_DEFINES
-#define _USE_MATH_DEFINES
+// Instead of using the constant from math.h or cmath we define it ourselves. This way
```
Comment (JonasVautherin) [inline core/global_include.h]: :+1:

File: plugins/camera/camera_impl.cpp [hunk 4]
```diff
@@ -509,8 +510,13 @@ Camera::Result CameraImpl::set_mode(const Camera::Mode mode)
...
 void CameraImpl::save_camera_mode(const float mavlink_camera_mode)
 {
+    if (!std::isfinite(mavlink_camera_mode)) {
+        LogWarn() << "Can't save NAN as camera mode";
+        return;
+    }
+
     MAVLinkParameters::ParamValue value;
-    value.set_uint32(mavlink_camera_mode);
+    value.set_uint32(uint32_t(mavlink_camera_mode));
```
Comment (JonasVautherin) [inline plugins/camera/camera_impl.cpp]: What about `static_cast<uint32_t>(mavlink_camera_mode)`? Looks more modern to me :sweat_smile:

Comment (julianoes) [reply plugins/camera/camera_impl.cpp]: True.

**Comment by julianoes** [comment]
I'm merging this and ignoring the jenkins CI for the branch. I think it fails because it greps for warning which is the name of the branch and therefore triggers when it should no.

**Pull request merged**